dao-voting/src/main.sw

```diff
@@ -0,0 +1,312 @@
+contract;
+
+dep abi;
+dep data_structures;
+dep errors;
+dep events;
+
+use std::{
+    address::Address,
+    assert::require,
+    block::height,
+    chain::auth::{AuthError, msg_sender},
+    context::{call_frames::msg_asset_id, msg_amount, this_balance},
+    contract_id::ContractId,
+    core::ops::Eq,
+    identity::Identity,
+    logging::log,
+    result::*,
+    revert::revert,
+    storage::StorageMap,
+    token::transfer,
+};
+
+use abi::DaoVoting;
+use data_structures::{Proposal, ProposalInfo};
+use errors::{CreationError, InitializationError, ProposalError, UserError};
+use events::{CreatePropEvent, DepositEvent, ExecuteEvent, UnlockVotesEvent, VoteEvent, WithdrawEvent};
+
+enum State {
+    NotInitialized: (),
+    Initialized: (),
+}
+
+impl Eq for State {
+    fn eq(self, other: Self) -> bool {
+        match(self, other) {
+            (State::Initialized, State::Initialized) => true, (State::NotInitialized, State::NotInitialized) => true, _ => false, 
+        }
+    }
+}
+
+storage {
+    // The amount of governance tokens a user has deposited
+    balances: StorageMap<Identity,
+    u64>, /// Contract Id of the governance token
+    gov_token: ContractId,
+    /// Information describing a proposal created via create_proposal(...)
+    proposals: StorageMap<u64,
+    ProposalInfo>, /// Number of created proposals
+    /// Used to check the validity of a proposal id
```

Comment:
hmm not sure that I would consider it a unique identifier.  In fact since I'm using 0 based "indexing" `proposal_count` should never refer to a `proposal_id`

Reply:
You're passing in an ID into functions which is directly used to reference a proposal based on the count. That is a unique identifier

Reply:
But the ID is the unique identifier.  `proposal_count` does not identify any proposals so I don't think it should be reffered to as a unique identifier.